proof.go

```diff
@@ -143,6 +143,13 @@ func (proof Proof) VerifyNamespace(h hash.Hash, nID namespace.ID, data [][]byte,
 	nIDLen := nID.Size()
 	if proof.IsOfAbsence() {
 		gotLeafHashes = append(gotLeafHashes, proof.leafHash)
+		// conduct some sanity checks:
+		leafMinNID := namespace.ID(proof.leafHash[:nIDLen])
+		if !nID.Less(leafMinNID) {
+			// leafHash.minNID  must be greater than nID
+			return false
+		}
```

Comment:
[nit] is it safe to remove min in this context? AFAIK a leaf can only have a singular namespace ID

Reply:
Can you please elaborate on your suggestion? Are you suggesting to change the naming of the variable `leafMinNID`?

Reply:
Sorry I meant something like 
```suggestion
		// conduct some sanity checks:
		leafNID := namespace.ID(proof.leafHash[:nIDLen])
		if !nID.Less(leafNID) {
			// leafHash.NID  must be greater than nID
			return false
		}
```

but we can ignore given your other comment